ibis/expr/operations/relations.py

```diff
@@ -785,3 +785,29 @@ class NotExistsSubquery(Node):
 
     def output_type(self):
         return ir.ExistsExpr
+
+
+@public
+class FillNa(TableNode, sch.HasSchema):
+    """Fill null values in the table."""
+
+    table = Arg(rlz.table)
+    value = Arg(rlz.any)
+    subset = Arg(rlz.list_of(rlz.string), default=[])
```

Comment:
This should probably be

```python
Arg(rlz.list_of(rlz.column_from("table")))
```

IIUC. It looks like you want to restrict `subset` to be a (possibly empty) list of column names from `table`. The way you have it now will allow any value that can be converted to a `StringValue`.

Reply:
If you end up accepting a dictionary here, you won't be able to use this rule. Instead you'll have to do something like:

```python
Arg(rlz.instance_of(collections.abc.Mapping))
```

and we can add a more thorough rule later. I'd suggest doing validation in the API functions for now, so that later when we add a proper rule, we are backwards compatible.

Reply:
Updated to use the `Arg(rlz.instance_of(collections.abc.Mapping))` validator + included manual validation in the API function that the columns exist in the table.